packages/test-worker/src/test-worker-instance.ts

```diff
@@ -87,13 +95,23 @@ export class TestWorkerInstance implements ITestWorkerInstance {
     public async execute(file: IFile, parameters: any, envParameters: any): Promise<any> {
         return new Promise(async (resolve, reject) => {
             try {
-                return await this.makeExecutionRequest(
-                    file,
-                    parameters,
-                    envParameters,
-                    resolve,
-                    reject
-                );
+                if (this.config.localWorker) {
+                    return await this.makeExecutionLocal(
```

Comment:
Maybe you should merge `makeExecutionLocal` and `makeExecutionRequest`? They have same API, inner implementations looks similar too, move local worker check inside this function.